okhttp-clients/src/main/java/com/palantir/remoting3/okhttp/InstrumentedInterceptor.java

```diff
@@ -32,46 +33,22 @@
 
     private static final Logger log = LoggerFactory.getLogger(InstrumentedInterceptor.class);
 
-    private final Meter informational;
-    private final Meter successful;
-    private final Meter redirection;
-    private final Meter clientError;
-    private final Meter serverError;
-    private final Meter other;
+    private final Map<String, HostMetrics> hostMetrics = Maps.newConcurrentMap();
+    private final MetricRegistry registry;
+    private final String name;
 
     InstrumentedInterceptor(MetricRegistry registry, String name) {
-        informational = registry.meter(MetricRegistry.name(name, "response", "family", "informational"));
-        successful    = registry.meter(MetricRegistry.name(name, "response", "family", "successful"));
-        redirection   = registry.meter(MetricRegistry.name(name, "response", "family", "redirection"));
-        clientError   = registry.meter(MetricRegistry.name(name, "response", "family", "client-error"));
-        serverError   = registry.meter(MetricRegistry.name(name, "response", "family", "server-error"));
-        other         = registry.meter(MetricRegistry.name(name, "response", "family", "other"));
+        this.registry = registry;
+        this.name = name;
     }
 
     @Override
     public Response intercept(Chain chain) throws IOException {
         Response response = chain.proceed(chain.request());
+        String hostName = chain.request().url().host();
 
-        switch (javax.ws.rs.core.Response.Status.Family.familyOf(response.code())) {
-            case INFORMATIONAL:
-                informational.mark();
-                break;
-            case SUCCESSFUL:
-                successful.mark();
-                break;
-            case REDIRECTION:
-                redirection.mark();
-                break;
-            case CLIENT_ERROR:
-                clientError.mark();
-                break;
-            case SERVER_ERROR:
-                serverError.mark();
-                break;
-            case OTHER:
-                other.mark();
-                break;
-        }
+        HostMetrics metrics = hostMetrics.computeIfAbsent(hostName, host -> new HostMetrics(registry, name, host));
+        metrics.update(response.code());
```

Comment:
think I'd prefer doing the per-host thing inside HostMetrics and make this call `record(response.code(), hostName)`